internal/page/donor/how_do_you_know_your_certificate_provider.go

```diff
@@ -40,14 +40,12 @@ func HowDoYouKnowYourCertificateProvider(tmpl template.Template, donorStore Dono
 					requireLength = true
 				}
 
-				if requireLength {
-					// TODO: should stay as Completed if editing and not changing the answer here
-					lpa.Tasks.CertificateProvider = actor.TaskInProgress
-				} else {
-					lpa.CertificateProvider.RelationshipLength = ""
-					lpa.Tasks.CertificateProvider = actor.TaskCompleted
+				if !requireLength {
+					lpa.CertificateProvider.RelationshipLength = 0
 				}
 
+				lpa.Tasks.CertificateProvider = actor.TaskInProgress
+
```

Comment:
Should we remove the address if they change how they know them? Otherwise we should only be making the task "in progress" here if they go pro->personal (as they have to answer how long they have known them)

Reply:
I hadn't thought about address but it feels right to start fresh if the relationship changes. And that makes sense, I think I'd oversimplified the existing logic when working out what should happen - I'll make it more explicit.